src/compile.ts

```diff
@@ -20,6 +20,7 @@ export async function compile(
   directory: string,
   ignore: string[] = [],
   solcModule: string = 'solc',
+  solcSettings: object = {optimizer: {enabled: true, runs: 200}},
```

Comment:
Running the optimizer by default will only add to the running time for no benefit since we are only interested in the AST.

```suggestion
  solcSettings: object = {},
```

Reply:
@frangio: That is solc's default, hence what most projects use by default.
So anyone experiencing the same issue as mine (compilation completes successfully when optimization is enabled, but aborts with a `Stack too deep` error when optimization is disabled) will have a really hard time understanding why they are able to compile their code, but unable to run `solidity-docgen` (which aborts with a `Stack too deep` error).
I believe that running-time is minor in this case (but it's your call of course).

Reply:
That being said, perhaps it would be better to set these two default values (for `solcModule` and for `solcSettings`) in file `cli.ts` instead. I did not quite understand how default-values initialization works in your system.

Reply:
>compilation completes successfully when optimization is enabled, but aborts with a Stack too deep error when optimization is disabled

That's odd... but if that's the case I'm okay with keeping these defaults.

I think it's okay to set these defaults in this file. I did it this way because the decision of what is the default value is encapsulated in this module.

Reply:
@frangio:
As I wrote in the PR description - this is exactly what I have experienced in [this contract](https://github.com/bancorprotocol/contracts/blob/0.5.0/solidity/contracts/BancorNetwork.sol).

Regarding the default-values issue, I was just concerned about what might happen if the user passed `solc-settings`(the 2nd argument of function `compile`)  but didn't pass `solc-module` (the 1st argument of function `compile`)... Wouldn't that yield function `compile` receiving `solcModule` as `undefined`?

Reply:
@frangio:
Just want to emphasize that I don't mind using your (empty) default value instead of my original suggestion. It will surely make the code (your code) look nicer, and in addition (as you've mentioned), will allow `solidity-docgen` run faster.

The `compilation completes successfully when optimization is enabled, but aborts with a Stack too deep error when optimization is disabled` scenario is probably not going to be very common, so execution time is indeed an incentive to consider in this case.

I'm going to pass `--solc-settings=JSON.stringify({optimizer: {enabled: true, runs: 200}})`, in my script whether we use it as a default value or not.
So please let me know if you still want me to commit your suggestion.

Thanks

Reply:
Let's keep the optimizer enabled by default. I don't think it will add too much compilation time and I want the default to work out of the box for people. I'm not sure how rare this issue is.